pkg/proxy/iptables/proxier.go

```diff
@@ -1395,37 +1409,25 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 	}
 
-	// Delete chains no longer in use. Since "iptables-save" can take several seconds
-	// to run on hosts with lots of iptables rules, we don't bother to do this on
-	// every sync in large clusters. (Stale chains will not be referenced by any
-	// active rules, so they're harmless other than taking up memory.)
+	// Delete chains no longer in use, if we previously got the list of existing chains
 	deletedChains := 0
-	if !proxier.largeClusterMode || time.Since(proxier.lastIPTablesCleanup) > proxier.syncPeriod {
-		var existingNATChains map[utiliptables.Chain]struct{}
-
-		proxier.iptablesData.Reset()
-		if err := proxier.iptables.SaveInto(utiliptables.TableNAT, proxier.iptablesData); err == nil {
-			existingNATChains = utiliptables.GetChainsFromTable(proxier.iptablesData.Bytes())
-
-			for chain := range existingNATChains {
-				if !activeNATChains[chain] {
-					chainString := string(chain)
-					if !isServiceChainName(chainString) {
-						// Ignore chains that aren't ours.
-						continue
-					}
-					// We must (as per iptables) write a chain-line
-					// for it, which has the nice effect of flushing
-					// the chain. Then we can remove the chain.
-					proxier.natChains.Write(utiliptables.MakeChainLine(chain))
-					proxier.natRules.Write("-X", chainString)
-					deletedChains++
+	if len(existingNATChains) > 0 {
```

Comment:
I was staring for a few minutes and I could not understand well the change, the commit says

> so that we have access to existingNATChains in the body of the loop.

but is only accessed here that is in the same place it was before, is this on preparation of something that will access later the rules in the body? or am I missing something?

Reply:
doh, didn't see there is an additional commit,